token/program-2022/src/extension/mod.rs

```diff
@@ -368,11 +368,36 @@ impl<'data, S: BaseState> StateWithExtensionsMut<'data, S> {
     }
 
     /// Packs the extension data into an open slot if not already found in the
-    /// data buffer, otherwise overwrites itself
+    /// data buffer
     pub fn init_extension<V: Extension>(&mut self) -> Result<&mut V, ProgramError> {
         self.init_or_get_extension(true)
     }
 
+    /// If `extension_type` is an Account-associated ExtensionType, this method packs the relevant
+    /// Extension of an ExtensionType into an open slot if not already found in the data buffer.
+    /// For all other ExtensionTypes, this is a no-op.
+    pub fn init_account_extension_from_type(
+        &mut self,
+        extension_type: ExtensionType,
+    ) -> Result<(), ProgramError> {
+        if extension_type.get_account_type() != AccountType::Account {
+            return Ok(());
+        }
+        match extension_type {
+            ExtensionType::TransferFeeAmount => {
+                self.init_extension::<TransferFeeAmount>().map(|_| ())
+            }
+            ExtensionType::ConfidentialTransferAccount => self
+                .init_extension::<ConfidentialTransferAccount>()
+                .map(|_| ()),
```

Comment:
Sorry, I didn't notice this on the first pass, but did we want confidential transfers to be only opt-in post-initialization?  I remember a discussion about not creating Confidential bits by default.

Reply:
Yes, I think that's correct, and it should be handled by `ExtensionType::get_account_extensions`. But I don't love having two methods that devs will have to remember to edit when implementing a new extension type. This is why I initially handled all extensions here.

Anyway, I've updated the ConfidentialTransfer case to be a no-op here, updated the name of `get_account_extensions` to be more explicit, and will ponder whether we can improve specifying "required on account initialization" across the two functions.